lib/logstash/filters/json.rb

```diff
@@ -75,11 +75,18 @@ def filter(event)
     end
 
     begin
-      # TODO(sissel): Note, this will not successfully handle json lists
-      # like your text is '[ 1,2,3 ]' json parser gives you an array (correctly)
-      # which won't merge into a hash. If someone needs this, we can fix it
-      # later.
-      dest.merge!(LogStash::Json.load(source))
+      parsed = LogStash::Json.load(source)
+      # If your parsed JSON is an array, we can't merge, so you must specify a
+      # destination to store the JSON, so you will get an exception about
+      if parsed.kind_of?(Java::JavaUtil::ArrayList)
```

Comment:
What's the advantage of `event.to_hash.merge! parsed` over `dest.merge!(parsed)`? Seems like we want to merge into dest in case dest is not the event itself.